diff --git a/service/java/com/android/server/wifi/SoftApManager.java b/service/java/com/android/server/wifi/SoftApManager.java
index 9e9f8126bb..3f94a75425 100644
--- a/service/java/com/android/server/wifi/SoftApManager.java
+++ b/service/java/com/android/server/wifi/SoftApManager.java
@@ -813,6 +813,16 @@ public class SoftApManager implements ActiveModeManager {
             // Absence of country code is not fatal for 2Ghz & Any band options.
             return true;
         }
+        if (!ApConfigUtil.isSoftApDynamicCountryCodeSupported(mContext)
+                || !mCurrentSoftApCapability.areFeaturesSupported(
+                        SoftApCapability.SOFTAP_FEATURE_ACS_OFFLOAD)) {
+            if (band == SoftApConfiguration.BAND_5GHZ) {
+                Log.e(getTag(), "Failed to set country code, "
+                        + "required for setting up soft ap in 5GHz");
+                return false;
+            }
+            return true;
+        }
         if (!mWifiNative.setApCountryCode(
                 mApInterfaceName, mCountryCode.toUpperCase(Locale.ROOT))) {
             if (band == SoftApConfiguration.BAND_5GHZ || band == SoftApConfiguration.BAND_6GHZ) {
