diff --git a/basic/non_plat/kernel.te b/basic/non_plat/kernel.te
index 5831906..40672a0 100644
--- a/basic/non_plat/kernel.te
+++ b/basic/non_plat/kernel.te
@@ -50,6 +50,8 @@ allow kernel misc2_block_device:blk_file rw_file_perms;
 allow kernel fuseblk:dir create_dir_perms;
 allow kernel fuseblk:file create_file_perms;
 
+allow kernel fuse:dir search;
+
 # Date : WK16.30
 # Operation: SQC
 # Purpose: Allow sdcardfs workqueue to access lower file systems
diff --git a/basic/non_plat/mnld.te b/basic/non_plat/mnld.te
index 996dea4..8baae27 100644
--- a/basic/non_plat/mnld.te
+++ b/basic/non_plat/mnld.te
@@ -119,3 +119,5 @@ allow mnld sysfs_leds:file r_file_perms;
 #Add for /nvcfg/almanac.dat
 allow mnld nvcfg_file:dir w_dir_perms;
 allow mnld nvcfg_file:file create_file_perms;
+
+get_prop(mnld, default_prop)
diff --git a/bsp/non_plat/file_contexts b/bsp/non_plat/file_contexts
index 8cb09fa..3c908d2 100644
--- a/bsp/non_plat/file_contexts
+++ b/bsp/non_plat/file_contexts
@@ -115,7 +115,6 @@
 # MTK VTService
 /dev/socket/volte_imsvt1(/.*)? u:object_r:volte_imsvt1_socket:s0
 
-/dev/goodix_fp     u:object_r:fingerprint_device:s0
 
 #MTK widevine kernel driver
 /dev/drm_wv u:object_r:widevine_drv_device:s0
